module-2/cdk/lib/CICDConstruct.ts

```diff
@@ -0,0 +1,127 @@
+import codebuild = require("@aws-cdk/aws-codebuild");
+import codecommit = require("@aws-cdk/aws-codecommit");
+import codepipeline = require("@aws-cdk/aws-codepipeline");
+import actions = require("@aws-cdk/aws-codepipeline-actions");
+import ecr = require("@aws-cdk/aws-ecr");
+import ecs = require("@aws-cdk/aws-ecs");
+import iam = require("@aws-cdk/aws-iam");
+import { ServicePrincipal } from "@aws-cdk/aws-iam";
+import cdk = require("@aws-cdk/cdk");
+import { EcrConstruct } from "./EcrConstruct";
+import { EcsClusterConstruct } from "./EcsConstruct";
+
+export class CiCdConstruct extends cdk.Construct {
+  
+  constructor(scope: cdk.Construct, id: string, ecsCluster: EcsClusterConstruct, ecrConstruct: EcrConstruct) {
+    super(scope, id);
+
+    const codeBuildStatement1 = new iam.PolicyStatement();
```

Comment:
Use `grant` methods (e.g. `bucket.grantRead`) instead of explicit IAM statements.